### PR TITLE
Move tests for `near-vm-logic` from `/tests` to `/src/tests/`

### PR DESCRIPTION
**Pull request by pmnoxx**
Move tests for `near-vm-logic` from `/tests` to `/src/tests`.

As part of #5371

### CODE REVIEW

**Comment by pmnoxx** [comment]
http://nayduck.near.org/#/run/2258

File: runtime/near-vm-logic/Cargo.toml [hunk 1]
```diff
@@ -43,29 +43,29 @@ costs_counting = []
...
 
 [[test]]
 name = "test_promises"
-path = "tests/test_promises.rs"
+path = "src/tests/test_promises.rs"
 
 [[test]]
 name = "test_iterators"
-path = "tests/test_iterators.rs"
+path = "src/tests/test_iterators.rs"
```
Comment (mina86) [inline runtime/near-vm-logic/Cargo.toml:71]: Shouldn’t those be deleted now?

**Comment by pmnoxx** [comment]
Running nayduck on expensive tests only: http://nayduck.near.org/#/run/2259

File: runtime/near-vm-logic/Cargo.toml (hunk 1 above)
Comment (pmnoxx) [reply runtime/near-vm-logic/Cargo.toml]: Thanks, good observation. I deleted those lines.

File: runtime/near-vm-logic/src/tests/mod.rs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+mod helpers;
+mod test_context;
+mod test_gas_counter;
+mod test_iterators;
+mod test_miscs;
+mod test_promises;
+mod test_registers;
+mod test_storage_read_write;
+mod test_storage_usage;
+mod test_view_method;
```
Comment (mina86) [inline runtime/near-vm-logic/src/tests/mod.rs:11]: Perhaps remove the `test` prefix?

**Review by mina86 — APPROVED** [review]
LGTM.

**Review by mm-near — APPROVED** [review]

**Comment by pmnoxx** [comment]
http://nayduck.near.org/#/run/2269

**Pull request merged**